src/Particle/ParticleSet.h

```diff
@@ -562,16 +562,8 @@ class ParticleSet : public QMCTraits, public OhmmsElementBase, public PtclOnLatt
   ///return the last index of a group i
   inline int last(int igroup) const { return SubPtcl[igroup + 1]; }
 
-  ///return the size of a group
-  inline int getGroupID(int iat) const
-  {
-    if (iat < 0 || iat >= TotalNum)
-      throw std::runtime_error("ParticleSet::getGroupID iat is out of range.");
-    int id = 0;
-    while (iat >= SubPtcl[id + 1])
-      id++;
-    return id;
-  }
+  ///return the group id of a given particle in the particle set.
+  inline int getGroupID(int iat) const { return GroupID[iat]; }
```

Comment:
Removed the error check to get inlining and good optimization?

Perhaps add an assert that would be active during DEBUG?

Reply:
Added assert which is effective in a debug build.